sdk/core/core-lro/src/lroEngine/operation.ts

```diff
@@ -0,0 +1,103 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT license.
+
+import { AbortSignalLike } from "@azure/abort-controller";
+import { PollOperationState, PollOperation } from "../pollOperation";
+import {
+  PollerConfig,
+  ResumablePollOperationState,
+  LongRunningOperation,
+  LroStatus
+} from "./models";
+import { getPollingUrl } from "./requestUtils";
+import { createInitializeState, createPollForLROStatus } from "./stateMachine";
+
+export class GenericPollOperation<TResult, TState extends PollOperationState<TResult>>
+  implements PollOperation<TState, TResult> {
+  private getLROStatusFromResponse?: (
+    pollingURL: string,
+    pollerConfig: PollerConfig
+  ) => Promise<LroStatus<TResult>>;
+  private pollerConfig?: PollerConfig;
+  constructor(
+    public state: TState & ResumablePollOperationState<TResult>,
+    private lro: LongRunningOperation<TResult>
+  ) {}
+
+  public setPollerConfig(pollerConfig: PollerConfig): void {
+    this.pollerConfig = pollerConfig;
+  }
+
+  /**
+   * General update function for LROPoller, the general process is as follows
+   * 1. Check initial operation result to determine the strategy to use
+   *  - Strategies: Location, Azure-AsyncOperation, Original Uri
+   * 2. Check if the operation result has a terminal state
+   *  - Terminal state will be determined by each strategy
+   *  2.1 If it is terminal state Check if a final GET request is required, if so
+   *      send final GET request and return result from operation. If no final GET
+   *      is required, just return the result from operation.
+   *      - Determining what to call for final request is responsibility of each strategy
+   *  2.2 If it is not terminal state, call the polling operation and go to step 1
+   *      - Determining what to call for polling is responsibility of each strategy
+   *      - Strategies will always use the latest URI for polling if provided otherwise
+   *        the last known one
+   */
+  async update(options?: {
```

Comment:
This is really the gem of all things LRO. We should either add a logger here that would showcase the behavior on debug, or add a guide on how to debug LROs that would point to how to add log lines and debugger instructions on this specific function.

Reply:
@xirzec I do not know much about logging, do you have ideas about whether I should do something special here to help have a good debugging experience?

Reply:
Perhaps to start, you can import `createClientLogger` from `@azure/logger` to make a logger for `core-lro`, then call `verbose` on it to log whats going on

Reply:
Done.